server/src/internalClusterTest/java/org/elasticsearch/index/shard/SearchIdleIT.java

```diff
@@ -187,6 +190,16 @@ public void testPendingRefreshWithIntervalChange() throws Exception {
         }
     }
 
+    private static void scheduleRefresh(IndexShard shard, boolean expectRefresh) {
+        PlainActionFuture<Boolean> future = PlainActionFuture.newFuture();
+        shard.scheduledRefresh(future);
+        if (expectRefresh) {
```

Comment:
nit: with `assertThat(..., equalTo(...))` there is no need for this if/else I think.